pkg/odo/cli/catalog/describe/component.go

```diff
@@ -0,0 +1,182 @@
+    %[1]s nodejs`)
+
+	componentLongDesc = ktemplates.LongDesc(`Describe a component type.
+This describes the component and its associated starter projects.
+`)
+)
+
+// DescribeComponentOptions encapsulates the options for the odo catalog describe component command
+type DescribeComponentOptions struct {
+	// name of the component to describe, from command arguments
+	componentName string
+	// if devfile components with name that matches arg[0]
+	devfileComponents []catalog.DevfileComponentType
+	// if componentName is a classic/odov1 component
+	component string
+	// generic context options common to all commands
+	*genericclioptions.Context
+}
+
+// NewDescribeComponentOptions creates a new DescribeComponentOptions instance
+func NewDescribeComponentOptions() *DescribeComponentOptions {
+	return &DescribeComponentOptions{}
+}
+
+// Complete completes DescribeComponentOptions after they've been created
+func (o *DescribeComponentOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.componentName = args[0]
+	if !pushtarget.IsPushTargetDocker() {
+		o.Context = genericclioptions.NewContext(cmd, true)
+		catalogList, err := catalog.ListComponents(o.Client)
+		if err != nil {
+			if experimental.IsExperimentalModeEnabled() {
+				klog.V(4).Info("Please log in to an OpenShift cluster to list OpenShift/s2i components")
+			} else {
+				return err
+			}
+		}
+		for _, image := range catalogList.Items {
+			if image.Name == o.componentName {
+				o.component = image.Name
+			}
+		}
+	}
+
+	if experimental.IsExperimentalModeEnabled() {
+		catalogDevfileList, err := catalog.ListDevfileComponents("")
+		if err != nil {
+			return err
+		}
+		for _, devfileComponent := range catalogDevfileList.Items {
+			if devfileComponent.Name == o.componentName {
+				o.devfileComponents = append(o.devfileComponents, devfileComponent)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Validate validates the DescribeComponentOptions based on completed values
+func (o *DescribeComponentOptions) Validate() (err error) {
+	if len(o.devfileComponents) == 0 && o.component == "" {
+		return errors.Wrapf(err, "No components with the name \"%s\" found", o.componentName)
+	}
+
+	return nil
+}
+
+// Run contains the logic for the command associated with DescribeComponentOptions
+func (o *DescribeComponentOptions) Run() (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+		if log.IsJSON() {
+			if len(o.devfileComponents) > 0 {
+				for _, devfileComponent := range o.devfileComponents {
+					data, err := pkgUtil.DownloadFileInMemory(devfileComponent.Registry.URL + devfileComponent.Link)
+					if err != nil {
+						return errors.Wrapf(err, "Failed to download devfile.yaml for devfile component: %s", devfileComponent.Name)
+					}
+					devObj, err := devfile.ParseInMemory(data)
+					if err != nil {
+						return err
+					}
+
```

Comment:
Should we abstract this out to something `getDevfile()` ?, as it is used in other condition as well.